### PR TITLE
please_cli: store nix and commit hash to index service on each deploy

### LINKED ISSUE #1641: store nix hash and revision to taskcluster index on data field for each deployment
1. we need to change each deploy task in [decision_task.py](https://github.com/mozilla/release-services/blob/master/lib/please_cli/please_cli/decision_task.py#L82) so that it will create an index with a nix hash and revision.

2. read more about index service api here: https://docs.taskcluster.net/docs/reference/core/taskcluster-index/references/api

especially read the part about **Indexing Routes**.

3. each deploy task should have the following route defined

```json
{
  "routes": [
    "index.project.releng.services.deployments.<channel>.<project>"
  ],
  "extra": {
    "index": {
      "data": {
        "nix_hash": "<nix_hash>",
        "revision": "<git_revision>"
      }
    }
}
```

4. `<nix_hash>` is calculated [here](https://github.com/mozilla/release-services/blob/master/lib/please_cli/please_cli/decision_task.py#L435) for each project. in code we call it `project_hash`

5. `<git_revision>` is passed in as an argument `github_commit`

6. `<channel>` is also passes and an argument `channel`

### PR DESCRIPTION
**Pull request by garbas**
fixes #1641

### CODE REVIEW

**Comment by garbas** [comment]
@La0 this is now ready to be merged, but i'd like you also to take a look if you have time.

File: lib/please_cli/please_cli/decision_task.py [hunk 1]
```diff
@@ -443,16 +472,65 @@ def cmd(ctx,
...
 
-    projects_to_deploy = []
+    click.echo(' => Checking github for project revisions')
+
+    project_revisions = {p: github_commit for p in PROJECTS}
+    project_url_cache = dict()
+    for project in sorted(PROJECTS):
+        # TODO: how should we handle outside projects?
+        #       skip project defined outside for now
+        if project in please_cli.config.OUTSIDE_PROJECTS:
```
Comment (La0) [inline lib/please_cli/please_cli/decision_task.py]: Maybe let the outside project specify a full taskcluster route name in their own config ?

File: lib/please_cli/please_cli/decision_task.py [hunk 2]
```diff
@@ -443,16 +472,65 @@ def cmd(ctx,
...
         click.echo(' => Checking which project needs to be redeployed')
 
-        # TODO: get status for our index branch
-        deployed_projects = {}
-
         for project_name in sorted(PROJECTS):
-            deployed_projects.get(project_name)
+            if project_name in deployed_projects and \
+               project_name in project_hashes and \
+               deployed_projects[project_name]['nix_hash'] == project_hashes[project_name]:
```
Comment (La0) [inline lib/please_cli/please_cli/decision_task.py]: :100:

Comment (La0) [reply lib/please_cli/please_cli/decision_task.py]: Nit: add a `log.debug` here saying that you skipped the deploy because of a nix hash match

**Pull request merged**